src/Aspire.Hosting.Azure/AzurePublisher.cs

```diff
@@ -0,0 +1,272 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Publishing;
+using Azure.Provisioning;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+using Azure.Provisioning.Resources;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Azure;
+
+internal sealed class AzurePublisher(
+    [ServiceKey] string name,
+    IOptionsMonitor<AzurePublisherOptions> options,
+    IOptions<AzureProvisioningOptions> provisioningOptions,
+    ILogger<AzurePublisher> logger) : IDistributedApplicationPublisher
+{
+    private AzureProvisioningOptions ProvisioningOptions => provisioningOptions.Value;
+
+    public Task PublishAsync(DistributedApplicationModel model, CancellationToken cancellationToken)
+    {
+        var publisherOptions = options.Get(name);
+
+        var infra = new Infrastructure
+        {
+            TargetScope = DeploymentScope.Subscription
+        };
+
+        var environmentParam = new ProvisioningParameter("environmentName", typeof(string));
+        infra.Add(environmentParam);
+
+        var locationParam = new ProvisioningParameter("location", typeof(string));
+        infra.Add(locationParam);
+
+        var principalId = new ProvisioningParameter("principalId", typeof(string));
+        infra.Add(principalId);
+
+        var tags = new ProvisioningVariable("tags", typeof(object))
+        {
+            Value = new BicepDictionary<string>
+            {
+                ["aspire-env-name"] = environmentParam
+            }
+        };
+
+        // REVIEW: Do we want people to be able to change this
```

Comment:
We could potentially provide an override. I think some people dislike the whole rg- thing (I do, it reminds me of warts on variable names).

But I think we should keep it as default for back compat.

Reply:
Via options for sure.